### ISSUE TITLE
[java] int64 in a path parameter is generated as Integer instead of Long

### ISSUE THREAD

**Issue by legart** [swagger-api/swagger-codegen-generators#219]
The following:

```json
       "/v1/{someLongValue}": {
            "get": {
                "summary": "My method summary",
                "operationId": "myMethodName",
                "produces": [
                    "application/json"
                ],
                "parameters": [
                    {
                        "name": "someLongValue",
                        "in": "path",
                        "description": "A description",
                        "required": true,
                        "type": "integer",
                        "format": "int64",
                        "x-example": 246470
                    }
                ],
```

generates 

```java
public ... myMethodName(Integer someLongValue) throws RestClientException {
```
with codegen version 3 - where version 2 would generate the correct

```java
public ... myMethodName(Long someLongValue) throws RestClientException {
```

In the model the Long values are correct, it seems to only affect path parameters.

This is using io.swagger.codegen.v3:swagger-codegen-cli:3.0.2

**Comment by mhombauersignon** [comment]
I'm having the exact same problem

**Comment by mhombauersignon** [comment]
To elaborate a bit more on this:
I'm using v3.0.5 and CXF REST client as target

**Comment by atokle** [comment]
I do also have the same problem, tried with 3.0.4 and 3.0.5, with feign as target

**Comment by frantuma** [comment]
@mhombauersignon @atokle can you please try with latest 3.0.6-SNAPSHOT (available on sonatype) and share some more details (failing spec, full command + options)?

**Comment by mhombauersignon** [comment]
3.0.6-SNAPSHOT seems to be working!

But anyway, here's the maven plugin configuration:
`
<plugin>
    <groupId>io.swagger.codegen.v3</groupId>
    <artifactId>swagger-codegen-maven-plugin</artifactId>
    <version>3.0.5</version>
    <executions>
        <execution>
            <goals>
                <goal>generate</goal>
            </goals>
            <configuration>
                <inputSpec>${project.basedir}/swagger/swagger.json</inputSpec>
                <ignoreFileOverride>${project.basedir}/.swagger-codegen-ignore</ignoreFileOverride>

                <language>jaxrs-cxf-client</language>

                <modelPackage>xxx.api.dto</modelPackage>
                <apiPackage>xxx.api.services</apiPackage>
                <invokerPackage>xxx.api</invokerPackage>
                <output>${project.build.directory}/generated-sources</output>

                <generateApiTests>false</generateApiTests>
                <generateApiDocumentation>false</generateApiDocumentation>
                <generateModelTests>false</generateModelTests>
                <generateModelDocumentation>false</generateModelDocumentation>
                <generateSupportingFiles>false</generateSupportingFiles>

                <withXml>false</withXml>
                <configOptions>
                    <dateLibrary>java8</dateLibrary>
                    <java8>true</java8>
                    <useGenericResponse>true</useGenericResponse>
                </configOptions>
            </configuration>
        </execution>
    </executions>
</plugin>
`

**Comment by atokle** [comment]
It works OK to me now with 3.0.6-SNAPSHOT

I noticed it also fixed parameter defined as
            "type": "string",
            "format": "date-time"
which in 3.0.5 resulted in String parameter, but with 3.0.6-SNAPSHOT is changed to OffsetDateTime

**Comment by mhombauersignon** [comment]
Confirmed as fixed in 3.0.7
